scipy/stats/_mstats_basic.py

```diff
@@ -88,6 +88,28 @@ def _chk_size(a, b):
     return (a, b, na)
 
 
+def _ttest_finish(df, t, alternative):
```

Comment:
This is just moved from `_stats_py`.

Reply:
Why there any reason not to unify this with `_get_pvalue` too? Is it because masked arrays aren't supported by the distribution infrasturcture?

Reply:
Yes. I guess the only problem is that it doesn't respect the mask. I guess we could pass the data in and then reapply the masks to the result, but I didn't bother.

Reply:
Is it as simple as that? If so, it would be good to get rid of the function completely!

Reply:
Well, I don't really want to add this functionality to `_get_pvalue` because it would adds some overhead that just isn't needed there. So we still need the function unless we cant to copy-paste similar code everywhere. Anyway, here is what it might look like; up to you whether to keep it.

Reply:
I see what you were saying. Look unclean to me, especially the arbitrary operations like `df + t`. I am happy with reverting the last commit and just adding a small comment on top of the `_ttest_finish` function saying something like "Can't unify this with the `_get_pvalue` function since the distribution infratructure doesn't support masked arrays."

Reply:
That was what the comment
> # We use ``stdtr`` directly here to preserve masked arrays

was for, which is already in the previous commit. If it's OK; I'll just revert and skip CI. Feel free to adjust the comment before merging.

Reply:
Ah Ok. I missed the comment. We can just revert then!